crates/pathfinder/src/storage/state.rs

```diff
@@ -51,28 +57,103 @@ impl L1StateTable {
                         :ethereum_transaction_index,
                         :ethereum_log_index
                     )",
-                named_params! {
-                    ":starknet_block_number": update.block_number.0,
-                    ":starknet_global_root": update.global_root.0.as_be_bytes(),
-                    ":ethereum_block_hash": &update.origin.block.hash.0[..],
-                    ":ethereum_block_number": update.origin.block.number.0,
-                    ":ethereum_transaction_hash": &update.origin.transaction.hash.0[..],
-                    ":ethereum_transaction_index": update.origin.transaction.index.0,
-                    ":ethereum_log_index": update.origin.log_index.0,
-                },
-            )
-            .context("Insert L1 state update")?;
+            named_params! {
+                ":starknet_block_number": update.block_number.0,
+                ":starknet_global_root": update.global_root.0.as_be_bytes(),
+                ":ethereum_block_hash": &update.origin.block.hash.0[..],
+                ":ethereum_block_number": update.origin.block.number.0,
+                ":ethereum_transaction_hash": &update.origin.transaction.hash.0[..],
+                ":ethereum_transaction_index": update.origin.transaction.index.0,
+                ":ethereum_log_index": update.origin.log_index.0,
+            },
+        );
+
+        match result {
+            Ok(_) => Ok(()),
+            Err(err) => {
+                if err.to_string().contains("A PRIMARY KEY constraint failed") {
+                    let latest = L1_LATEST_REORG
+                        .lock()
+                        .map_err(|_| anyhow::anyhow!("L1 accounting lock is poisoned"))
+                        .context("Aquire L1 accounting lock")?;
+
+                    tracing::error!(pre_reorg=%latest.0, post_reorg=%latest.1, count=%latest.2, "Additional L1 reorg bug information");
```

Comment:
I am unsure if we would still be able to but I would like to query for the conflicting row. At least the rusqlite::ErrorCode::ConstraintViolation description includes "abort" which I would expect to deny all commands in this transaction, but I haven't tested it.

Reply:
I've added the query and log the row. I tested it with an in-memory db and it worked fine despite the aborting.